### ISSUE TITLE
BioMistral Generating \n repeatedly after training

### ISSUE THREAD

**Issue by DurgaSandeep25** [huggingface/trl#1434]
Hi, I am using BioMistral to finetune my data and a data collator(completion only LM) to compute loss only on the response. However, after training, it mainly generates \n most of the time. But giving a simple extra token like "The" generates much better than before. Why is the model so sensitive? Is there a way to avoid generating "\n" tokens? Following is my code

def formatting_prompts_inference_new(sample):

    input_text = sample["dialogue"]
    category = sample['section_header']
    text = "### Instruction: Summarize the following conversation based on the category\n ###Category: " + category +"\n ### Conversation:\n" + input_text + "\n ### Summary:\n"
    sample['prompt'] = text

    return sample

I fine-tuned with the above prompt, but the only change is after " ### Summary:\n", I have the target summary while training. Happy to provide more details if required. thanks!

**Comment by DurgaSandeep25** [comment]
I have tried bad_word_ids to not generate "\n". But I am still not able to understand why the SFT trained model is generating that way (raw BioMistral is working fine)

**Comment by younesbelkada** [comment]
Hi @DurgaSandeep25 
Hmm it's a bit hard to tell, are you using a custom data collator by any chance? This might indicate a silent bug during your training
